### ISSUE TITLE
Water replication bug

### ISSUE THREAD

**Issue by Zhid13** [CleverRaven/Cataclysm-DDA#32625]
# Describe the bug

When trying to pick up a non reseleable container with water, in my case a teapot, the game asks wether to wield the container or pour the water somewhere else/consume it. When trying to consume it, water will be consumed, but no charges will be lost off the container. 

# Steps To Reproduce

1. Place non resealable container with water in tile/Fill adjacent container with water
2. Try to pick up container. 
3. Select to drink the water when asked what to do. 

# Expected behavior

I expected the water to be consumed from the container.

# Versions and configuration

 - OS: Windows 7
 - Game Version: 0.D-5733-g457d06c
 - Graphics version: Tiles
 - Mods loaded: 
dda
C.R.I.T
Fict guns
Icecoons arsenal
GF recipes
Craft ammo
ERG
Craft ammo - ERG
Garden pots
Gun Cotton
Makeshift items
More survival tools
Packaging
Craftable gun pack
Draco's dog mod
Modular turrets
Hydroponics
Mining mod
More city locations
Folding parts
Tanks and other vehicles
Arcana
sees-player icon -attitude
Artyom's gun emporium reloaded
Extended buildings
Safe autodoc
Bionic slots
Stats through skills
Manual bionic installation
Classes and scenarios
Mutation changes
SpeedyDex
Stats through kills
Extra proffesions
No bees
No ants
No explosive zombies
Wintertime classes
Cataclysm ++

**Comment by ProfoundDarkness** [comment]
TLDR; confirmed.

A quick test with just no npc needs mod and after trying to pickup the nearby teapot, selecting spill and then consume, the player's need for water is reduced and the water in the item (teapot) remains unchanged.

This only seems to happen when trying to pick up-spill-consume.  If you wield the teapot then try to unwield it (back to inventory), on selecting spill, consume, 1 unit of the teapot is consumed properly.

**Comment by ipcyborg** [comment]
Reason:

pickup.cpp:
>// We already checked in do_pickup if this was a nullptr
>// Make copies so the original remains untouched if we bail out
>item newit = *loc.get_item();

and then:
>picked_up = newit.spill_contents( u );

But it will return false (not all liquid is consumed). And the consumption is done from the copy (newit).
That also explains why it works fine if all the liquid is consumed.

Suggested fix:
>picked_up = loc.get_item()->spill_contents( u );

I have tested it, looks fine. 
Will make a PR for it.
